docs/modules/ROOT/pages/maploader-and-mapstore.adoc

```diff
@@ -1,96 +1,171 @@
 
 {description}
 
-MapStore supports the following caching patterns:
+MapStore builds a cache on top of Hazelcast with defined triggers for operations on a map. You can build a MapStore using either of the following module classes, depending on your requirements:
 
-- *Read-through:* If an entry does not exist in memory when an application asks for it, Hazelcast asks the MapStore implementation to load that entry from the data store. If the entry exists, the MapStore implementation requests it from the data store and Hazelcast puts it into memory.
-- *Write-through:* When new entries are added to a map, those entries are added to the data store synchronously.
-- *Write-behind:* When new entries are added to a map, those entries are added to the data store asynchronously, after a configured delay.
+- MapLoader allows you to load data to your cluster from external systems. It supports read-through caching only.
+- MapStore allows you to extend the functionality offered by MapLoader to store and delete data in your external systems. This ensures that all your data stores are synchronized. It supports read-through, write-through, and write-behind caching.
 
-MapStore extends the MapLoader interface with the methods to store and delete data in the external data source, whereas MapLoader contains the methods to load data into Hazelcast. You can use either of these interfaces to meet your requirements, but this guide refers to both interfaces as _MapStore_.
+The caching patterns are as follows:
 
-The following steps describe the workflow for developing a MapStore:
+[cols="h,2,4"]
+|===
+|Caching pattern |Supported by |Description
 
-. <<implement, Implement>>
-. <<test, Test>>
-. <<deploy, Deploy>>
-. <<configure, Configure>>
+|Read-through
+|MapLoader and MapStore
+|If the requested entry is not present in memory, MapStore requests it from the data store. 
 
-[[implement]]
-== Implementing a MapStore
+If present in the data store, the entry is loaded to the distributed map and remains in memory until removed.
 
-To implement a MapStore for {hazelcast-cloud}, you can mostly xref:hazelcast:data-structures:working-with-external-data.adoc[follow the Hazelcast Platform documentation]. However, you must make sure to manage the lifecycle of the MapStore. When the cluster loads the MapStore, it makes a connection to the external data store. If open connections are left open, these connections will consume unnecessary resources. To manage this connection, make sure to implement the link:https://docs.hazelcast.org/docs/latest/javadoc/com/hazelcast/map/MapLoaderLifecycleSupport.html[`MapLoaderLifeCycleSupport` interface] to allow the cluster to close connections to the data store on a graceful shutdown.
+|Write-through
+|MapStore only
+|Adds entries to the data store synchronously when added to a map.
 
-[[test]]
-== Testing a MapStore
+|Write-behind
+|MapStore only
+|Adds entries to the data store asynchronously after a configured delay when added to a map.
```

Comment:
This wording makes it look like the delay is intentional. 

But that's not the case - there are two reasons to store the data async
1. to not block the thread updating the map
2. to collect more updates into a single batch of updates

the configured delay puts a limit on the delay between the write to the map and write to the database, it can be lower.

Reply:
Very useful, thank you. I have expanded the description